packages/eslint-plugin/lib/util.js

```diff
@@ -127,3 +127,17 @@ exports.getParserServices = context => {
   }
   return context.parserServices;
 };
+
+/**
+ * @template T
+ * @param {ReadonlyArray<T> | undefined} a An array
+ * @param {ReadonlyArray<T> | undefined} b Another array
+ * @param {function(T, T): boolean} eq Comparison function
+ * @returns {boolean} Returns true iff the arrays are equal
```

Comment:
honk iff you like formal logic

Reply:
HOOOOONK

sadly, this was lifted straight from the TSLint implementation.
I've removed it, since it was only for back-compat purposes (specifically, TS 2.5).